src/smefit/analyze/pca.py

```diff
@@ -1,16 +1,109 @@
+        self.coefficients = coefficients
+        self.config = config
+
+    @classmethod
+    def from_dict(cls, config):
+
+        loaded_datasets = load_datasets(
+            config["data_path"],
+            config["datasets"],
+            config["coefficients"],
+            config["order"],
+            config["use_quad"],
+            config["use_theory_covmat"],
+            config["use_t0"],
+            config.get("use_multiplicative_prescription", False),
+            config.get("theory_path", None),
+            config.get("rot_to_fit_basis", None),
+            config.get("uv_coupligs", False),
+        )
+
+        coefficients = CoefficientManager.from_dict(config["coefficients"])
+
+        return cls(loaded_datasets, coefficients, config)
+
+    def build_runcard(self):
+
+        pca = PcaCalculator(self.loaded_datasets, self.coefficients, None)
+        pca.compute()
+
+        # dump the rotation matrix
+        # the roation matrix is composed by two blocks: PCA and an identity for the constrained dofs
+        fixed_dofs = self.coefficients.name[~self.coefficients.is_free]
+        id_df = pd.DataFrame(
+            np.eye(fixed_dofs.size), columns=fixed_dofs, index=fixed_dofs
+        )
+        rotation = pd.concat([pca.pc_matrix, id_df]).replace(np.nan, 0)
+
+        rot_dict = {
+            "name": "PCA_rotation",
+            "xpars": rotation.index.tolist(),
+            "ypars": rotation.columns.tolist(),
+            "matrix": rotation.values.tolist(),
+        }
+        rot_mat_path = pathlib.Path(
+            self.config["result_path"], self.config["result_ID"], "pca_rot.json"
+        )
+
+        with open(rot_mat_path, "w") as f:
+            json.dump(rot_dict, f)
+
+        self.config["rot_to_fit_basis"] = str(rot_mat_path)
+
+        # update coefficient contraints
+        self.coefficients.update_constrain(rotation.T)
+        new_coeffs = {}
+        for coef_obj in self.coefficients._objlist:
+            if coef_obj.constrain is None:
+                continue
+            if "value" in self.config["coefficients"][coef_obj.name]:
+                new_coeffs[coef_obj.name] = self.config["coefficients"][coef_obj.name]
+            if coef_obj.constrain is not None:
+                tmp = {
+                    "min": coef_obj.minimum,
+                    "max": coef_obj.maximum,
+                    "constrain": coef_obj.constrain,
+                }
+                new_coeffs[coef_obj.name] = tmp
+        for pc in pca.pc_matrix.columns:
+            # TODO: rotate prio volume correctly
+            new_coeffs[pc] = {"min": -5, "max": 5}
+        self.config["coefficients"] = new_coeffs
+
+    def save(self):
+        p = pathlib.Path(self.config["result_path"])
+        root_path = pathlib.Path(*p.parts[:-1])
+        self.config["pca_rotation"] = False
+
+        result_ID = self.config["result_ID"]
+        runcard_copy = root_path / "runcards" / f"{result_ID}.yaml"
```

Comment:
As it is right now it overwrite the original `runcard`, which is good but can also be a problem, we need to improve this.